CalibCalorimetry/EBPhase2TPGTools/plugins/EcalEBPhase2TPParamProducer.cc

```diff
@@ -0,0 +1,337 @@
+#include "Geometry/CaloGeometry/interface/CaloCellGeometry.h"
+#include "Geometry/Records/interface/CaloGeometryRecord.h"
+#include "Geometry/EcalMapping/interface/EcalElectronicsMapping.h"
+#include "Geometry/EcalMapping/interface/EcalMappingRcd.h"
+#include "DataFormats/EcalDigi/interface/EcalConstants.h"
+#include "DataFormats/EcalDetId/interface/EcalSubdetector.h"
+#include "DataFormats/EcalDetId/interface/EBDetId.h"
+
+#include "CalibCalorimetry/EBPhase2TPGTools/plugins/EcalEBPhase2TPParamProducer.h"
+
+#include <iostream>
+#include <string>
+#include <sstream>
+#include <vector>
+#include <ctime>
+#include <fstream>
+#include <iomanip>
+
+EcalEBPhase2TPParamProducer::EcalEBPhase2TPParamProducer(edm::ParameterSet const& pSet)
+    : theBarrelGeometryToken_(esConsumes(edm::ESInputTag("", "EcalBarrel"))),
+      inFile_(pSet.getUntrackedParameter<std::string>("inputFile")),
+      outFile_(pSet.getUntrackedParameter<std::string>("outputFile")),
+      nSamplesToUse_(pSet.getParameter<unsigned int>("nSamplesToUse")),
+      useBXPlusOne_(pSet.getParameter<bool>("useBXPlusOne")),
+      phaseShift_(pSet.getParameter<double>("phaseShift")),
+      nWeightGroups_(pSet.getParameter<unsigned int>("nWeightGroups")),
+      theEcalTPGPedestals_Token_(esConsumes(edm::ESInputTag("", ""))),
+      et_sat_(pSet.getParameter<double>("Et_sat")),
+      xtal_LSB_(pSet.getParameter<double>("xtal_LSB")),
+      binOfMaximum_(pSet.getParameter<unsigned int>("binOfMaximum"))
+
+{
+  out_file_ = gzopen(outFile_.c_str(), "wb");
+
+  const TString* inFileName = new TString(inFile_);
+  TFile* inFile = new TFile(*inFileName, "READ");
```

Comment:
```suggestion
  TFile* inFile = new TFile(inFile_.c_str(), "READ");
```
can avoid new'ing the `inFileName`

Reply:
Ok